hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/compact/SparkScheduleCompactionActionExecutor.java

```diff
@@ -65,10 +66,12 @@ protected HoodieCompactionPlan scheduleCompaction() {
 
     int deltaCommitsSinceLastCompaction = table.getActiveTimeline().getDeltaCommitTimeline()
         .findInstantsAfter(lastCompactionTs, Integer.MAX_VALUE).countInstants();
-    if (config.getInlineCompactDeltaCommitMax() > deltaCommitsSinceLastCompaction) {
+    if (config.getInlineCompactDeltaCommitMax() > deltaCommitsSinceLastCompaction
+                    && timeCompaction(instantTime, initialTime, deltaCommitsSinceLastCompaction)) {
```

Comment:
how about triggering a compaction when any one of the conditions is met?

Reply:
yes. So two conditions in `if` means they won't trigger compact.
Return true means imcompactable.